File: vscode/samurai-agent/tests/agent/code_parser/fixtures/python/sample_types.py
# Python Type Definitions Test Fixture
# Tests type_definition element extraction

from typing import TypeAlias, Union, List, Dict

# PEP 613 Type Aliases
UserId: TypeAlias = str
UserRole: TypeAlias = str

# Generic type aliases
Point: TypeAlias = tuple[float, float]
Matrix: TypeAlias = List[List[float]]

# Union types
Result: TypeAlias = Union[int, str, None]

# Complex type alias
ApiResponse: TypeAlias = Dict[str, Union[str, int, List[str]]]

# Protocol (structural typing)
from typing import Protocol

class Drawable(Protocol):
    def draw(self) -> None:
        ...

class Closeable(Protocol):
    def close(self) -> None:
        ...

File: vscode/samurai-agent/tests/agent/code_parser/fixtures/typescript/sample_types.ts
// TypeScript Type Definitions Test Fixture
// Tests type_definition element extraction

export type UserId = string;
export type UserRole = 'admin' | 'user' | 'guest';

type Point = {
    x: number;
    y: number;
};

export type GenericResponse<T> = {
    data: T;
    status: number;
    message: string;
};

// Interface for comparison
export interface User {
    id: UserId;
    name: string;
    role: UserRole;
}

// Type alias with generics
export type ApiResult<T, E = Error> = 
    | { success: true; data: T }
    | { success: false; error: E };

